### PR TITLE
Fix a bug in array index conversion

### PR DESCRIPTION
**Pull request by clover2123**
Signed-off-by: HyukWoo Park <hyukwoo.park@samsung.com>

### CODE REVIEW

File: test/cctest/testapi.cpp [hunk 1]
```diff
@@ -344,6 +344,18 @@ TEST(ValueRef, Basic2) {
     EXPECT_EQ(ii, 123);
 }
 
+TEST(ValueRef, Basic3) {
+    Evaluator::execute(g_context.get(), [](ExecutionStateRef* state) -> ValueRef* {
+        auto value = ValueRef::create(-1);
```
Comment (bbrto21) [inline test/cctest/testapi.cpp:349]: It would be nice if you add more tests using variety values!

Comment (clover2123) [reply test/cctest/testapi.cpp:349]: Thanks for the review. I added one more test which checks the maximum value of uint32_t.

**Review by bbrto21 — APPROVED** [review]
LGTM!

**Review by ksh8281 — APPROVED** [review]
